include/pstl_offload/internal/usm_memory_replacement.h

```diff
@@ -59,28 +60,49 @@ class __offload_policy_holder_type
     // of the class is inline, we need to avoid calling static functions inside of the constructor
     // and pass the pointer to exact function as an argument to guarantee that the correct __active_device
     // would be stored in each translation unit
-    __offload_policy_holder_type(sycl::device&& __offload_device,
+    template <typename _DeviceSelector>
+    __offload_policy_holder_type(const _DeviceSelector& __device_selector,
                                  __set_active_device_func_type __set_active_device_func)
-        : _M_offload_device(std::move(__offload_device)), _M_offload_policy(_M_offload_device),
-          _M_set_active_device(__set_active_device_func)
+        : _M_set_active_device(__set_active_device_func)
     {
-        _M_set_active_device(&_M_offload_device);
+        try
+        {
+            _M_offload_device.emplace(__device_selector);
+            _M_set_active_device(&_M_offload_device.value());
```

Comment:
```suggestion
            _M_set_active_device(&*_M_offload_device);
```
I apologize. I changed my mind because I found out the technical reason, why `*` is better. It's unchecked version and never throws. I didn't know it previously.

Reply:
Done.